### ISSUE TITLE
File names clash under Windows

### ISSUE THREAD

**Issue by pattacini** [robotology/yarp#2691]
**Describe the bug**

The following files generate names clash under case-insensitive OS like Windows:

  `src/libYARP_dev/src/idl_generated_code/yarp/dev/AudioPlayerStatus_common.h`
  `src/libYARP_dev/src/idl_generated_code/yarp/dev/audioPlayerStatus_common.h`
  `src/libYARP_dev/src/idl_generated_code/yarp/dev/AudioRecorderStatus_common.h`
  `src/libYARP_dev/src/idl_generated_code/yarp/dev/audioRecorderStatus_common.h`

**To Reproduce**
Just clone YARP anew on Windows and you'll get:

```console
$ git clone https://github.com/robotology/yarp.git
Cloning into 'yarp'...
remote: Enumerating objects: 188299, done.
remote: Counting objects: 100% (10627/10627), done.
remote: Compressing objects: 100% (4523/4523), done.
remote: Total 188299 (delta 6047), reused 9888 (delta 5765), pack-reused 177672
Receiving objects: 100% (188299/188299), 89.63 MiB | 4.76 MiB/s, done.
Resolving deltas: 100% (135372/135372), done.
Updating files: 100% (6937/6937), done.
warning: the following paths have collided (e.g. case-sensitive paths
on a case-insensitive filesystem) and only one from the same
colliding group is in the working tree:

  'src/libYARP_dev/src/idl_generated_code/yarp/dev/AudioPlayerStatus_common.h'
  'src/libYARP_dev/src/idl_generated_code/yarp/dev/audioPlayerStatus_common.h'
  'src/libYARP_dev/src/idl_generated_code/yarp/dev/AudioRecorderStatus_common.h'
  'src/libYARP_dev/src/idl_generated_code/yarp/dev/audioRecorderStatus_common.h'
```

**Expected behavior**
We should avoid name clash from upfront as on Windows Git operates in such a way that it removes locally one group of files leaving the working tree modified.

**Configuration (please complete the following information):**
 - OS: Windows 10
 - yarp version: `master`
-  compiler: not relevant

**Additional context**

---

cc @randaz81 @drdanz

**Comment by pattacini** [comment]
@randaz81 @drdanz any news on this?

**Comment by pattacini** [comment]
Fixed via #2699 ✔️
